src/quote/flashmint/hyeth/component-quotes/morpho.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable  @typescript-eslint/no-non-null-assertion */
+import { BigNumber } from '@ethersproject/bignumber'
+import { Contract } from '@ethersproject/contracts'
+
+import { WETH } from 'constants/tokens'
+import { SwapQuoteProvider } from 'quote/swap'
+import { isSameAddress } from 'utils/addresses'
+import { getRpcProvider } from 'utils/rpc-provider'
+
+export class MorphoQuoteProvider {
```

Comment:
In the future we might make this into a more generic quote provider since a couple of providers use this token vault standard [4626](https://eips.ethereum.org/EIPS/eip-4626) but for now waiting what components will be added.